ext/ddtrace_profiling_native_extension/libdatadog_helpers.c

```diff
@@ -0,0 +1,73 @@
+#include "libdatadog_helpers.h"
+
+#include <ruby.h>
+
+const char *ruby_value_type_to_string(enum ruby_value_type type) {
+  switch (type) {
+    case(RUBY_T_NONE    ): return "T_NONE";
+    case(RUBY_T_OBJECT  ): return "T_OBJECT";
+    case(RUBY_T_CLASS   ): return "T_CLASS";
+    case(RUBY_T_MODULE  ): return "T_MODULE";
+    case(RUBY_T_FLOAT   ): return "T_FLOAT";
+    case(RUBY_T_STRING  ): return "T_STRING";
+    case(RUBY_T_REGEXP  ): return "T_REGEXP";
+    case(RUBY_T_ARRAY   ): return "T_ARRAY";
+    case(RUBY_T_HASH    ): return "T_HASH";
+    case(RUBY_T_STRUCT  ): return "T_STRUCT";
+    case(RUBY_T_BIGNUM  ): return "T_BIGNUM";
+    case(RUBY_T_FILE    ): return "T_FILE";
+    case(RUBY_T_DATA    ): return "T_DATA";
+    case(RUBY_T_MATCH   ): return "T_MATCH";
+    case(RUBY_T_COMPLEX ): return "T_COMPLEX";
+    case(RUBY_T_RATIONAL): return "T_RATIONAL";
+    case(RUBY_T_NIL     ): return "T_NIL";
+    case(RUBY_T_TRUE    ): return "T_TRUE";
+    case(RUBY_T_FALSE   ): return "T_FALSE";
+    case(RUBY_T_SYMBOL  ): return "T_SYMBOL";
+    case(RUBY_T_FIXNUM  ): return "T_FIXNUM";
+    case(RUBY_T_UNDEF   ): return "T_UNDEF";
+    case(RUBY_T_IMEMO   ): return "T_IMEMO";
+    case(RUBY_T_NODE    ): return "T_NODE";
+    case(RUBY_T_ICLASS  ): return "T_ICLASS";
+    case(RUBY_T_ZOMBIE  ): return "T_ZOMBIE";
+    #ifdef RUBY_T_MOVED // Introduced in Ruby 2.7
+    case(RUBY_T_MOVED   ): return "T_MOVED";
+    #endif
+                  default: return "T_UNKNOWN_OR_MISSING_RUBY_VALUE_TYPE_ENTRY";
```

Comment:
Is this a default created for this PR? I couldn't find it the ruby's source code.

If so, I'd say keep it different from the others, and I find it quite hard to parse such a long screaming case string.

I suggest something like `"Unknown Ruby Value"`, or something that will unmistakably read as "Ha, this is not a valid Ruby VM type".

Reply:
Changed in d73a72f9527b9c5751f98b06a76e45189c080aed to `BUG: Unknown value for ruby_value_type`. As you suspected, the only objective here was to have an obvious wrong value that we could grep in the codebase and tie back to this function, so we can make it look even more as an obvious wrong value.